internal/featuredetection/feature_detection_test.go

```diff
@@ -40,10 +40,11 @@ func TestPullRequestFeatures(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			fakeHTTP := &httpmock.Registry{}
-			httpClient := api.NewHTTPClient(api.ReplaceTripper(fakeHTTP))
+			reg := &httpmock.Registry{}
+			httpClient := http.DefaultClient
```

Comment:
Let's never mutate `http.DefaultClient` and instead always construct a new `&http.Client{}`